### PR TITLE
Fix unit test bootstrap_processor.lazy_pruning_missing_block

### PR DESCRIPTION
**Pull request by dsiganos**

### CODE REVIEW

File: nano/core_test/bootstrap.cpp [hunk 1]
```diff
@@ -1306,6 +1306,12 @@ TEST (bootstrap_processor, lazy_pruning_missing_block)
 	config.frontiers_confirmation = nano::frontiers_confirmation_mode::disabled;
 	config.enable_voting = false; // Remove after allowing pruned voting
 	nano::node_flags node_flags;
+	// It looks like force confirmed elections remain in the AEC and
+	// they keep publishing blocks, disabling the request loop stops this.
+	// A better fix would be to create a way to add and cement the blocks
+	// without involving an election at all or have a way to delete an
+	// election from the AEC.
+	node_flags.disable_request_loop = true;
```
Comment (pwojcikdev) [inline nano/core_test/bootstrap.cpp]: This comment suggests that inserting and confirming blocks directly in the ledger `node->ledger.process` and `node->ledger.confirm` would be a better solution.

Comment (dsiganos) [reply nano/core_test/bootstrap.cpp]: If we have a way to insert and confirm without creating an election then that might be a cleaner solution, if I understood what is happening well.
I was not aware of `nano::ledger::confirm`.
I will try it later.

Comment (clemahieu) [reply nano/core_test/bootstrap.cpp]: For reference it was recently added here https://github.com/nanocurrency/nano-node/pull/4519

File: nano/secure/ledger.hpp [hunk 2]
```diff
@@ -116,6 +116,8 @@ class ledger final
 	// Returns the next receivable entry equal or greater than 'key'
 	std::optional<std::pair<nano::pending_key, nano::pending_info>> receivable_lower_bound (secure::transaction const & tx, nano::account const & account, nano::block_hash const & hash) const;
 	void initialize (nano::generate_cache_flags const &);
+
+public: // for unit tests only
```
Comment (pwojcikdev) [inline nano/secure/ledger.hpp]: This is mainly used by confirming set, so the comment is inaccurate.

Comment (dsiganos) [reply nano/secure/ledger.hpp]: I mean making the member function public for use in unit tests only.
I will make it more explicit.

Comment (pwojcikdev) [reply nano/secure/ledger.hpp]: This is still less than ideal. A safe pattern similar to the way it's done in other classes should be used that provides a public `force_confirm` method with a proper release_assert that guards it.

Comment (pwojcikdev) [reply nano/secure/ledger.hpp]: Let's follow good patterns that are already established and work. A single assert is more descriptive than any comment.

**Review by pwojcikdev — APPROVED** [review]

**Pull request merged**